sirepo/pkcli/job_agent.py

```diff
@@ -55,14 +57,48 @@ def default_command():
     pkdlog('{}', cfg)
     i = tornado.ioloop.IOLoop.current()
     d = _Dispatcher()
-    def s(n, x):
-        return i.add_callback_from_signal(d.terminate)
+    def s(*args):
+        return i.add_callback_from_signal(_terminate, d)
     signal.signal(signal.SIGTERM, s)
     signal.signal(signal.SIGINT, s)
     i.spawn_callback(d.loop)
     i.start()
 
 
+def start_sbatch():
+    def kill_agent(pid_file):
+        if socket.getfqdn() == pid_file.fqdn:
+            os.kill(pid_file.pid, signal.SIGKILL)
+        else:
+            try:
+                subprocess.run(
+                    ('ssh', pid_file.fqdn, 'kill', '-KILL', str(pid_file.pid)),
+                    capture_output=True,
+                ).check_returncode()
+            except subprocess.CalledProcessError as e:
```

Comment:
I think there will be an error if the process isn't found, which is a case we should ignore. The process crashed, for example, before it could remove the pidfile

Reply:
I've implemented just checking stderr for 'no such process'. Are you aware of anything more clever?

Reply:
For logging purposes, this is fine. If we get a system that starts outputting, say, "process does not exist", we can add a filter for that. Having extra logging isn't terrible, just makes it harder to see the real problems.